lib/app/services/api_network.dart

```diff
@@ -0,0 +1,70 @@
+import 'package:flutter/widgets.dart';
+import 'package:get_it/get_it.dart';
+import 'package:meta/meta.dart';
+import 'package:http/http.dart' as http;
+import 'package:schulcloud/app/app.dart';
+
+import 'storage.dart';
+
+/// A service that offers making network request to the backend servers. If the
+/// user's token is stored in the authentication storage, the requests' headers
+/// are automatically enriched with the access token.
+@immutable
+class ApiNetworkService {
```

Comment:
Please cherry-pick d2e8206 onto this branch to fix JSON-encoding of request bodies